worker/base/worker.go

```diff
@@ -245,10 +245,14 @@ func (w *Worker) Labels() map[string]string {
 
 func (w *Worker) Platforms(noCache bool) []ocispecs.Platform {
 	if noCache {
+		matchers := make([]platforms.MatchComparer, len(w.WorkerOpt.Platforms))
+		for i, p := range w.WorkerOpt.Platforms {
+			matchers[i] = platforms.Only(p)
```

Comment:
Although the effect of this isn't that significant compared to qemu detection logic, calling `platforms.Only` isn't completely free either as it internally allocates memory for the variant vector arrays.